CustomSkillForDataIngestion/QnAIntegrationCustomSkill/QnAIntegrationCustomSkill.cs

```diff
@@ -189,36 +209,44 @@ private static async Task<Operation> MonitorOperation(IQnAMakerClient qnaClient,
         // </MonitorOperation>
 
         // Checks valid file type before sending for extraction
-        private static bool IsValidFile(string fileName)
+        private static bool IsValidFile(string fileName, long fileSizeInKBs)
         {
-            HashSet<string> fileExtensions = new HashSet<string> { "tsv", "pdf", "txt", "docx", "xlsx" };
-            string fileExtension;
-            try
+            var fileExtension = Path.GetExtension(fileName)?.ToLower()?.TrimStart('.');
+            // Prior file type and size validation for QnA Maker
+            if (FileTypeToSizeLimitInQnAMaker.TryGetValue(fileExtension, out var szLimitInKb)
+                    && fileSizeInKBs <= szLimitInKb)
             {
-                fileExtension = Path.GetExtension(fileName).ToLower().TrimStart('.');
+                return true;
             }
-            catch
-            {
-                fileExtension = string.Empty;
-            }
-            
-            return fileExtensions.Contains(fileExtension);
+            return false;
         }
 
         // Returns the operation state after extraction for fileName
         private static string GetOperationStatus(Operation operation, string fileName, ILogger log)
         {
             string operationState = OperationStateType.Succeeded;
```

Comment:
Should operationState be set to Failed by default?  I guess it doesn't really matter since you are explicitly setting it in each case.

Reply:
I changed the code to a more readable one. In the else the default value is succeeded and it is set to failed only if the filename is present in the error response.